docker/Dockerfile

```diff
@@ -229,19 +224,19 @@ RUN set -e; \
     truncate -s0 ~/.bash_history;
 
 RUN set -e; \
-    if [ -z $(which bazel) ]; then \
+    if ! which bazel >/dev/null ; then \
         BAZEL_INSTALL_DIR='/usr/local/bin/'; \
         BAZEL_VERSION='5.2.0'; \
         BAZEL_INSTALL_PATH="$BAZEL_INSTALL_DIR/bazel-$BAZEL_VERSION"; \
         BAZEL_ARCH=$(case "$TARGETARCH" in \
-                     'amd64') \
-                         echo 'x86_64'; \
-                         ;; \
-                     'arm64') \
-                         echo 'arm64'; \
-                         ;; \
-                   esac); \
-        sudo wget -O "$BAZEL_INSTALL_PATH" "https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-linux-$BAZEL_ARCH"; \
+        'amd64') \
+            echo 'x86_64'; \
+            ;; \
+        'arm64') \
+            echo 'arm64'; \
+            ;; \
+        esac); \
+        sudo curl -o "$BAZEL_INSTALL_PATH" "https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-linux-$BAZEL_ARCH"; \
```

Comment:
Why sudo?
1. You should sudo tee >/dev/null in this case, not sudo curl
2. Always use 2-stage downloading, tmp first and them move. Use install to automatically set +x for you.